packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -44,8 +77,9 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
         latestSummary?: SummaryNode,
         initialSummary?: IInitialSummary,
         wipSummaryLogger?: ITelemetryLogger,
-        private readonly getGCDataFn?: () => Promise<IGCData>,
-        private readonly getInitialGCDataFn?: () => Promise<IGCData | undefined>,
+        private readonly getGCDataFn?: () => Promise<IGarbageCollectionData>,
+        getInitialGCDetailsFn?: () => Promise<IGarbageCollectionDetails>,
+        public usedRoutes: string[] = [],
```

Comment:
Is this publicly settable just for testing purposes? I don't see where this gets set, is it part of the follow-up PR?

Reply:
It would have happened in this PR - https://github.com/microsoft/FluidFramework/pull/4670. But a lot has changed since then so it has to be updated. Actually I am planning to add those changes here itself so its easier to look at the complete picture.

Reply:
So I should look for another commit to this PR?

Reply:
yes. I am working on it :-)

Reply:
Added the commit. Note that the work to update channel context's used routes is not yet done. After GC has run and generated used routes, the channel's used routes needs to be updated in 2 cases:
1.The data store is not realized so channel context has not been created. The used routes needs to be passed to the channel during creation.
2. The channel context exists in which case the routes can be passed to it right away.

This will happen when we start supporting GC at DDS layer. For now, a channel context is always considered used so its used routes contains itself only.